pkg/services/sqlstore/migrations/folder_mig.go

```diff
@@ -0,0 +1,36 @@
+package migrations
+
+import (
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func addFolderMigrations(mg *migrator.Migrator) {
+	mg.AddMigration("create folder table", migrator.NewAddTableMigration(folderv1()))
+
+	// copy any existing folders in the dashboard table into the new folder
+	// table. The *legacy* parent folder ID, stored as folder_id  in the
+	// dashboard table, is always going to be "0" so it is safe to convert to a parent UID.
+	mg.AddMigration("copy existing folders from dashboard table", migrator.NewRawSQLMigration(
+		"INSERT INTO folder (id, uid, org_id, title, parent_uid, created, updated) SELECT id, uid, org_id, title, folder_id, created, updated FROM dashboard WHERE is_folder = 1;",
+	).Postgres("INSERT INTO folder (id, uid, org_id, title, parent_uid, created, updated) SELECT id, uid, org_id, title, folder_id, created, updated FROM dashboard WHERE is_folder = true;"))
+
+	mg.AddMigration("Add index for parent_uid", migrator.NewAddIndexMigration(folderv1(), &migrator.Index{
+		Cols: []string{"parent_uid", "org_id"}, Type: migrator.UniqueIndex,
```

Comment:
Are we sure this would be a unique index? Many folders in the same org may share a parent.

Reply:
Good catch, thank you, definitely not!